compiler/rustc_save_analysis/src/lib.rs

```diff
@@ -899,28 +899,25 @@ impl<'a> DumpHandler<'a> {
 
     fn output_file(&self, ctx: &SaveContext<'_>) -> (BufWriter<File>, PathBuf) {
         let sess = &ctx.tcx.sess;
-        let file_name = match ctx.config.output_file {
-            Some(ref s) => PathBuf::from(s),
+        let mut root_path = match self.odir {
```

Comment:
Why is this changing here? Maybe I misunderstood something, but I'm not quite following the changes to this part of the code.

Reply:
The `let file_name = match ctx.config.output_file` logic will no longer work. `ctx.config.output_file` will never be `None` and will always contain the `build_base` dir (see why in the bullets below). Instead we check if there is an `output dir` and assign it to `root_path`, otherwise we assign the `build_base` dir. The `file_name` logic then names the output file to be the name of the running crate with a `.json` extension. In the previous code, if output dir is not set, the resulting `file_name` is always `a.json` thus overwriting the output files in `save-analysis-temp` if more than one `save-analysis` test where there is no `--out-dir` argument runs. 

ctx.config.output_file is always `build_base` dir for the following reasons:
* the output file (-o) and output dir (--out-dir) arguments are passed to `rustc` here where I also set the `RUST_SAVE_ANALYSIS_CONFIG`:
https://github.com/rust-lang/rust/blob/d8b033cdb3832a10d1e61f3c1f01ee1643ec5abc/src/tools/compiletest/src/runtest.rs#L1996-L2016 
* but only the output dir is ever passed to `rustc_save_analysis` where `config` is passed in as `None`:
https://github.com/rust-lang/rust/blob/d8b033cdb3832a10d1e61f3c1f01ee1643ec5abc/compiler/rustc_driver/src/lib.rs#L376-L387
* so the `find_config` function below will always result in `config.output_file` containing the `build_base` dir since the `config` passed to the function is always `None`: 
https://github.com/rust-lang/rust/blob/d8b033cdb3832a10d1e61f3c1f01ee1643ec5abc/compiler/rustc_save_analysis/src/lib.rs#L994-L1012